src/m3ninx/index/segment/mem/reader.go

```diff
@@ -45,6 +45,15 @@ type reader struct {
 	closed bool
 }
 
+// TODO(nate):
```

Comment:
I think this is only used in tests, so not super concerned about this impl at the moment